pkg/ranger/grpc.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package ranger
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+
+	"storj.io/storj/examples/piecestore/rpc/client/api"
+	"storj.io/storj/internal/pkg/readcloser"
+
+	pb "storj.io/storj/examples/piecestore/rpc/protobuf"
+)
+
+type grpcRanger struct {
+	client pb.RouteGuideClient
+	hash   string
+	size   int64
+}
+
+// GRPCRanger turns a gRPC connection into a Ranger
+func GRPCRanger(client pb.RouteGuideClient, hash string) (Ranger, error) {
+	shard, err := api.ShardMetaRequest(client, hash)
+	if err != nil {
+		return nil, err
+	}
+	return &grpcRanger{client, hash, shard.Size}, nil
+}
+
+// GRPCRangerSize creates a GRPCRanger with known size.
+// Use it if you know the piece size. This will safe the extra request for
+// retrieving the piece size from the piece storage.
+func GRPCRangerSize(client pb.RouteGuideClient, hash string, size int64) Ranger {
+	return &grpcRanger{client, hash, size}
```

Comment:
silly little style nitpick. if someone changes the grpcRanger struct later (adds a new field, reorders things, etc), sometimes this can surprisingly work even when it shouldn't. the best thing to do is always name your fields

Reply:
Good point. Fixed.